mod-di-converter-storage-server/src/main/java/org/folio/services/migration/ProfileMigrationServiceImpl.java

```diff
@@ -0,0 +1,194 @@
+package org.folio.services.migration;
+
+import io.micrometer.core.instrument.util.StringUtils;
+import io.vertx.core.Context;
+import io.vertx.core.Future;
+import io.vertx.core.Promise;
+import org.apache.commons.io.IOUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.folio.dao.PostgresClientFactory;
+import org.folio.dao.ProfileDao;
+import org.folio.dao.association.ProfileWrapperDao;
+import org.folio.dao.snapshot.ProfileSnapshotDao;
+import org.folio.dao.snapshot.ProfileSnapshotItem;
+import org.folio.okapi.common.GenericCompositeFuture;
+import org.folio.rest.impl.util.OkapiConnectionParams;
+import org.folio.rest.jaxrs.model.JobProfile;
+import org.folio.rest.jaxrs.model.JobProfileCollection;
+import org.folio.rest.jaxrs.model.ProfileAssociation;
+import org.folio.rest.jaxrs.model.ProfileSnapshotWrapper;
+import org.folio.rest.jaxrs.model.ProfileType;
+import org.folio.rest.jaxrs.model.ReactToType;
+import org.folio.rest.persist.PostgresClient;
+import org.folio.services.association.CommonProfileAssociationService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+import static java.lang.String.format;
+
+@Service
+public class ProfileMigrationServiceImpl implements ProfileMigrationService {
+  private static final Logger LOGGER = LogManager.getLogger();
+  private static final String UPDATE_WRAPPERS_INSIDE_ASSOCIATIONS = "templates/db_scripts/associations-migration/update_wrappers_inside_associations.sql";
+  private static final String REVERT_VIEW = "templates/db_scripts/associations-migration/revert_associations_view.sql";
+  private static final String UPDATE_SCHEMA_FOR_MIGRATION = "templates/db_scripts/associations-migration/actualize_schema_for_migrations.sql";
+  private static final String TENANT_PLACEHOLDER = "${myuniversity}";
+  private static final String MODULE_PLACEHOLDER = "${mymodule}";
+  @Autowired
+  protected PostgresClientFactory pgClientFactory;
+  @Autowired
+  private ProfileDao<JobProfile, JobProfileCollection> jobProfileDao;
+  @Autowired
+  private ProfileWrapperDao profileWrapperDao;
+  @Autowired
+  private CommonProfileAssociationService commonProfileAssociationService;
+  @Autowired
+  private ProfileSnapshotDao profileSnapshotDao;
+
+  @Override
+  public Future<Boolean> migrateDataImportProfiles(Map<String, String> headers, Context context) {
+    Promise<Boolean> creationProfilesFuture = Promise.promise();
+    String tenantId = new OkapiConnectionParams(headers).getTenantId();
+    return profileWrapperDao.checkIfDataInTableExists(tenantId)
+      .compose(e -> {
```

Comment:
nitpick: 'e' seems to be an error. something like 'isDataPresent' could work better.